src/bsoncxx/types.hpp

```diff
@@ -106,7 +108,7 @@ struct BSONCXX_API b_double {
 /// @relatesalso b_double
 ///
 BSONCXX_INLINE bool operator==(const b_double& lhs, const b_double& rhs) {
-    return lhs.value == rhs.value;
+    return fabs(lhs.value - rhs.value) < DBL_EPSILON;
```

Comment:
Recommend change: use the Standard Library's epsilon from std::numeric_limits. (Although I'm not entirely sure what the guarantees made by b_double are-- perhaps the macro is more appropriate for some internal reason, although that would surprise me.)
https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon

Reply:
OK, switched to using  std::numeric_limits<double>::epsilon(). I also realized that the epsilon needs to be scaled to the order of magnitude of the numbers, so I copied Donald Knuth's solution for that. :-)
https://stackoverflow.com/a/253874/10743364